cli/azd/pkg/project/scaffold_gen_environment_variables.go

```diff
@@ -556,6 +556,34 @@ func getResourceConnectionEnvs(usedResource *ResourceConfig,
 		default:
 			return []scaffold.Env{}, unsupportedAuthTypeError(resourceType, authType)
 		}
+	case ResourceTypeJavaEurekaServer:
+		return []scaffold.Env{
+			{
+				Name:  "eureka.client.register-with-eureka",
+				Value: "true",
+			},
+			{
+				Name:  "eureka.client.fetch-registry",
+				Value: "true",
+			},
+			{
+				Name:  "eureka.instance.prefer-ip-address",
+				Value: "true",
+			},
+			{
+				// Use '\${}' to escape parsing ${} placeholder in Bicep, just treat it as an Env of application
+				Name:  "eureka.client.serviceUrl.defaultZone",
+				Value: fmt.Sprintf("\\${%s_BASE_URL}/eureka", strings.ToUpper(usedResource.Name)),
+			},
+		}, nil
+	case ResourceTypeJavaConfigServer:
+		return []scaffold.Env{
+			{
+				// Use '\${}' to escape parsing ${} placeholder in Bicep, just treat it as an Env of application
+				Name:  "spring.config.import",
+				Value: fmt.Sprintf("optional:configserver:\\${%s_BASE_URL}", strings.ToUpper(usedResource.Name)),
```

Comment:
Same here,  move to `ToBicepEnv` in `bicep_env.go`.

Reply:
Move this escape logic to `ServiceConfigEnv to Scaffold Env`

Reply:
Same here.